release-notes/7.0/preview/api-diff/preview4/Microsoft.NETCore.App/7.0-preview4_System.Numerics.md

```diff
@@ -0,0 +1,198 @@
++        TSelf Sinh(TSelf x);
++        TSelf Tanh(TSelf x);
++    }
++    public interface IIncrementOperators<TSelf> where TSelf : IIncrementOperators<TSelf> {
++        TSelf op_CheckedIncrement(TSelf value);
++        TSelf operator ++(TSelf value);
++    }
++    public interface ILogarithmicFunctions<TSelf> where TSelf : ILogarithmicFunctions<TSelf> {
++        TSelf Log(TSelf x);
++        TSelf Log(TSelf x, TSelf newBase);
++        TSelf Log10(TSelf x);
++        TSelf Log2(TSelf x);
++    }
++    public interface IMinMaxValue<TSelf> where TSelf : IMinMaxValue<TSelf> {
++        TSelf MaxValue { get; }
++        TSelf MinValue { get; }
++    }
++    public interface IModulusOperators<TSelf, TOther, TResult> where TSelf : IModulusOperators<TSelf, TOther, TResult> {
++        TResult operator %(TSelf left, TOther right);
++    }
++    public interface IMultiplicativeIdentity<TSelf, TResult> where TSelf : IMultiplicativeIdentity<TSelf, TResult> {
++        TResult MultiplicativeIdentity { get; }
++    }
++    public interface IMultiplyOperators<TSelf, TOther, TResult> where TSelf : IMultiplyOperators<TSelf, TOther, TResult> {
++        TResult op_CheckedMultiply(TSelf left, TOther right);
++        TResult operator *(TSelf left, TOther right);
++    }
++    public interface INumber<TSelf> : IAdditionOperators<TSelf, TSelf, TSelf>, IAdditiveIdentity<TSelf, TSelf>, IComparable, IComparable<TSelf>, IComparisonOperators<TSelf, TSelf>, IDecrementOperators<TSelf>, IDivisionOperators<TSelf, TSelf, TSelf>, IEqualityOperators<TSelf, TSelf>, IEquatable<TSelf>, IFormattable, IIncrementOperators<TSelf>, IModulusOperators<TSelf, TSelf, TSelf>, IMultiplicativeIdentity<TSelf, TSelf>, IMultiplyOperators<TSelf, TSelf, TSelf>, INumberBase<TSelf>, IParsable<TSelf>, ISpanFormattable, ISpanParsable<TSelf>, ISubtractionOperators<TSelf, TSelf, TSelf>, IUnaryNegationOperators<TSelf, TSelf>, IUnaryPlusOperators<TSelf, TSelf> where TSelf : INumber<TSelf> {
++        TSelf Abs(TSelf value);
++        TSelf Clamp(TSelf value, TSelf min, TSelf max);
++        TSelf CopySign(TSelf value, TSelf sign);
++        TSelf CreateChecked<TOther>(TOther value);
++        TSelf CreateSaturating<TOther>(TOther value);
++        TSelf CreateTruncating<TOther>(TOther value);
++        bool IsNegative(TSelf value);
++        TSelf Max(TSelf x, TSelf y);
++        TSelf MaxMagnitude(TSelf x, TSelf y);
++        TSelf Min(TSelf x, TSelf y);
++        TSelf MinMagnitude(TSelf x, TSelf y);
++        TSelf Parse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider);
++        TSelf Parse(string s, NumberStyles style, IFormatProvider? provider);
++        int Sign(TSelf value);
++        bool TryCreate<TOther>(TOther value, out TSelf result);
++        bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider, out TSelf result);
++        bool TryParse(string? s, NumberStyles style, IFormatProvider? provider, out TSelf result);
```

Comment:
Notable from other diffs `CreateChecked`, `CreateSaturating`, `CreateTruncating`, and `TryCreate` are missing the generic constraint on `TOther`
```suggestion
+        static abstract TSelf Abs(TSelf value);
+        static abstract TSelf Clamp(TSelf value, TSelf min, TSelf max);
+        static abstract TSelf CopySign(TSelf value, TSelf sign);
+        static abstract TSelf CreateChecked<TOther>(TOther value) where TOther : INumber<TOther>;
+        static abstract TSelf CreateSaturating<TOther>(TOther value) where TOther : INumber<TOther>;
+        static abstract TSelf CreateTruncating<TOther>(TOther value) where TOther : INumber<TOther>;
+        static abstract bool IsNegative(TSelf value);
+        static abstract TSelf Max(TSelf x, TSelf y);
+        static abstract TSelf MaxMagnitude(TSelf x, TSelf y);
+        static abstract TSelf Min(TSelf x, TSelf y);
+        static abstract TSelf MinMagnitude(TSelf x, TSelf y);
+        static abstract TSelf Parse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider);
+        static abstract TSelf Parse(string s, NumberStyles style, IFormatProvider? provider);
+        static abstract int Sign(TSelf value);
+        static abstract bool TryCreate<TOther>(TOther value, out TSelf result) where TOther : INumber<TOther>;
+        static abstract bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider, out TSelf result);
+        static abstract bool TryParse(string? s, NumberStyles style, IFormatProvider? provider, out TSelf result);
```